ha/util/consul_kv_store.py

```diff
@@ -154,13 +197,31 @@ def get(self, key: str = ""):
         Return:
             str: Return dictionary of all key val pair.
         """
-        _, data = self._consul.kv.get(self._prepare_key(key), recurse=True)
-        if data is None:
-            return data
+        k = self._prepare_key(key)
+
+        # simulating consul kv get operation for _payload dict
+        payload_data = None
+        if self._enable_batch_put:
+            # Using dictionary comprehension + startswith()
+            # Prefix key match in dictionary
+            Log.debug(f"[consul-op] searching key in cache: {k}")
+            payload_data = {key:val for key, val in self._payload.items() if key.startswith(k)}
```

Comment:
If you get payload data, then you don't need to go ahead for consul kv get. right?
You can return payload_data. directly?

Reply:
Ok..Got it. Thanks!